assisted_deployment.sh

```diff
@@ -4,28 +4,241 @@ set -o pipefail
 source logging.sh
 source utils.sh
 source common.sh
-source assisted_deployment_utils.sh
 
 
+# Default values are managed through the operator itself.
+# The defaults are not being set here on purpose so that the
+# default operator flow can be used instead.
+ASSISTED_SERVICE_IMAGE="${ASSISTED_SERVICE_IMAGE:-}"
+ASSISTED_INSTALLER_IMAGE="${ASSISTED_INSTALLER_IMAGE:-}"
+ASSISTED_AGENT_IMAGE="${ASSISTED_AGENT_IMAGE:-}"
+ASSISTED_DATABASE_IMAGE="${ASSISTED_DATABASE_IMAGE:-}"
+ASSISTED_CONTROLLER_IMAGE="${ASSISTED_CONTROLLER_IMAGE:-}"
+ASSISTED_OPENSHIFT_VERSIONS="${ASSISTED_OPENSHIFT_VERSIONS:-}"
 
-function install_assisted_env() {
-  pushd $WORKING_DIR
-  if cd $TEST_INFRA_DIR;then
-    git fetch --all && git reset --hard origin/$TEST_INFRA_BRANCH;
-  else
-    git clone --branch $TEST_INFRA_BRANCH https://github.com/openshift/assisted-test-infra.git;
-  fi
-  popd
-  pushd $TEST_INFRA_DIR
-  KUBECONFIG=$PWD/minikube_kubeconfig ./create_full_environment.sh
-  popd
+ASSISTED_NAMESPACE="${ASSISTED_NAMESPACE:-assisted-installer}"
+ASSISTED_OPERATOR_INDEX="${ASSISTED_OPERATOR_INDEX:-quay.io/ocpmetal/assisted-service-index:latest}"
 
-  if ! [ "$MANAGE_BR_BRIDGE" == "y" ];then
-    switch_to_internal_dns
-  fi
+
+function deploy_local_storage() {
+  oc adm new-project openshift-local-storage || true
+
+  oc annotate project openshift-local-storage openshift.io/node-selector=''
+
+  cat <<EOF | oc apply -f -
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: local-operator-group
+  namespace: openshift-local-storage
+spec:
+  targetNamespaces:
+    - openshift-local-storage
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: local-storage-operator
+  namespace: openshift-local-storage
+spec:
+  installPlanApproval: Automatic
+  name: local-storage-operator
+  source: redhat-operators
+  sourceNamespace: openshift-marketplace
+EOF
+
+  wait_for_crd "localvolumes.local.storage.openshift.io"
+
+  echo "Creating local volume and storage class..."
+  cat <<EOCR | oc apply -f -
+apiVersion: local.storage.openshift.io/v1
+kind: LocalVolume
+metadata:
+  name: assisted-service
+  namespace: openshift-local-storage
+spec:
+  logLevel: Normal
+  managementState: Managed
+  storageClassDevices:
+    - devicePaths:
+        - /dev/sdb
```

Comment:
Where are the extra disks being created for the LocalVolume?

Reply:
Right now, manually! 

dev-scripts has an option to set `VM_EXTRADISKS` to true which will create one disk. This is not enough, so @lranjbar will work on a PR to allow dev-scripts for creating a custom number of disks that we can then use here.

This comment does raise a good point. Right now these disks are not configurable and it requires modifying this file manually. Wonder if it's worth making it configurable in this PR or just wait for @lranjbar PR